src/passes/StackCheck.cpp

```diff
@@ -130,9 +119,17 @@ struct StackCheck : public Pass {
       return;
     }
 
-    // Pick appropriate names.
-    auto stackBaseName = Names::getValidGlobalName(*module, "__stack_base");
-    auto stackLimitName = Names::getValidGlobalName(*module, "__stack_limit");
+    // __stack_base and __stack_end are created by the toolchain (emscripten)
+    // If we don't find these globals, error out.
+    auto stackBase = module->getGlobalOrNull("__stack_base");
+    if (!stackBase) {
+      Fatal() << "cannot do stack checking without __stack_base variable\n";
+    }
+
+    auto stackEnd = module->getGlobalOrNull("__stack_end");
+    if (!stackEnd) {
+      Fatal() << "cannot do stack checking without __stack_end variable\n";
+    }
```

Comment:
IIRC we decided that we should not add dependency the name section, but instead using the import/export names to find these?

I think we need to check for both exports (in the case of normal/main modules) and imports (in the case of side modules).

e.g.
```
auto stackEnd = module->getExportOrNull("__stack_end");
if (!stackEnd) {
   // somehow look for a global that was imported using the name `__stack_end`
}
```

Out of interest, does this current approach pass all the emscripten tests? i.e. are we already always including a name section?

Reply:
It looks like we don't even generate a name section under all circumstances: https://github.com/emscripten-core/emscripten/blob/0e75ea8f916f9b4baaa35a17824606836ffc0dc1/tools/building.py#L286-L292

We could add STACK_CHECK to this condition, but maybe better to use import/export names?